### PR TITLE
Str-617: fn-tests remove DEFAULT_ROLLUP_PARAMS in favor of RollupParams from datatool

### PR DESCRIPTION
**Pull request by evgenyzdanovich**
## Description

Remove DEFAULT_ROLLUP_PARAMS in favor of RollupParams from datatool

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactor
- [ ] New or updated tests
- [ ] Dependency Update

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code where necessary.
- [x] I have updated the documentation if needed.
- [x] My changes do not introduce new warnings.
- [ ] I have added tests that prove my changes are effective or that my feature works.
- [x] New and existing tests pass with my changes.

### CODE REVIEW

File: functional-tests/fn_bridge_deposit_reclaim_drt_seen.py [hunk 1]
```diff
@@ -147,7 +143,8 @@ def main(self, ctx: flexitest.RunContext):
             btc_password,
         )
         self.debug(f"User BTC balance (after takeback): {refund_btc_balance}")
-        expected_balance = 5 * DEPOSIT_AMOUNT - TAKE_BACK_FEE
+        deposit_amount = ctx.env.rollup_cfg().deposit_amount
```
Comment (storopoli) [inline functional-tests/fn_bridge_deposit_reclaim_drt_seen.py:146]: Maybe we should normalize like below?

```suggestion
        cfg: RollupConfig = ctx.env.rollup_cfg()
        deposit_amount = cfg.deposit_amount
```

File: functional-tests/fn_sync_bitcoin_reorg.py [hunk 2]
```diff
@@ -63,15 +61,17 @@ def main(self, ctx: flexitest.RunContext):
...
-def check_nth_checkpoint_finalized_on_reorg(checkpt_idx: int, seq: Service, btc: Service):
+def check_nth_checkpoint_finalized_on_reorg(
+    ctx: flexitest.RunContext, checkpt_idx: int, seq: Service, btc: Service
+):
     # Now submit another checkpoint proof and produce a couple of blocks(less than reorg depth)
     seqrpc = seq.create_rpc()
     btcrpc = btc.create_rpc()
     seq_addr = seq.get_prop("address")
-    finality_depth = FAST_BATCH_ROLLUP_PARAMS["l1_reorg_safe_depth"]
+    finality_depth = ctx.env.rollup_cfg().l1_reorg_safe_depth
```
Comment (storopoli) [inline functional-tests/fn_sync_bitcoin_reorg.py:74]: Same:

`cfg: RollupConfig = ctx.env.rollup_cfg()` then `cfg.foo`

File: functional-tests/rollup_params_cfg.py [hunk 3]
```diff
@@ -48,3 +48,13 @@ class RollupConfig(BaseModel):
     proof_publish_mode: ProofPublishMode
     max_deposits_in_block: int
     network: str
+
+    # Additional fields that aren't coming from datatool config generation (yet)
+    # and has to be supplied manually.
+    # TODO: extend datatool to return OPERATOR_FEE from bridge-tx-builder/src/constants.rs
```
Comment (storopoli) [inline functional-tests/rollup_params_cfg.py:54]: We should open a ticket, or we'll forget about this.

**Review by storopoli — CHANGES_REQUESTED** [review]
Much cleaner, thanks!
Some nits, mostly for consistency sake.

**Review by delbonis — COMMENTED** [review]
This is fantastic.  Agree with Jose's comments, overall looks good.

File: functional-tests/fn_bridge_deposit_reclaim_drt_seen.py (hunk 1 above)
Comment (evgenyzdanovich) [reply functional-tests/fn_bridge_deposit_reclaim_drt_seen.py]: Agree, done

File: functional-tests/fn_sync_bitcoin_reorg.py (hunk 2 above)
Comment (evgenyzdanovich) [reply functional-tests/fn_sync_bitcoin_reorg.py]: Agree, done

File: functional-tests/rollup_params_cfg.py (hunk 3 above)
Comment (evgenyzdanovich) [reply functional-tests/rollup_params_cfg.py]: Agree, done

**Review by delbonis — APPROVED** [review]

**Review by storopoli — APPROVED** [review]
ACK d98f1844e9a53fed8cd38ac452080132b78456f4

**Pull request merged**